src/network_engine.cpp

```diff
@@ -290,9 +290,9 @@ NetworkEngine::requestStep(Sp<Request> sreq)
         node.setExpired();
         if (not node.id)
             requests.erase(req.tid);
+        if (req.on_expired)
+            req.on_expired(req, false);
         return;
-    } else if (req.attempt_count == 1) {
-        req.on_expired(req, false);
```

Comment:
Calling on_expired with false is an intended optimisation to allow searches to flag a node as suspicious after the first request timeout, to allow completing the search without waiting for the node to expire. If the node replies before search completion, the flag is cleared and the answer is processed normally. If the node expires, all of its pending requests also expire, and on_expired is called with true.

Reply:
/* Time for a request to timeout */
    static constexpr const std::chrono::seconds MAX_RESPONSE_TIME {1};

I think 1 second timeout for sending a request is too small, it is inefficient on a network environment.